### ISSUE TITLE
t/Makefile.am: non-POSIX variable name

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-sched#403]
```
→ ./autogen.sh 
Running autoreconf -i ... 
t/Makefile.am:22: warning: shell if [ $(FLUX_PREFIX: non-POSIX variable name
t/Makefile.am:22: (probably a GNU make extension)
Cleaning up ...
Now run ./configure.
```

**Comment by dongahn** [comment]
Thank you for creating a ticket for this. I have meant to get to this for a LONG time. My first attempt is to use the `AM_CONDITIONAL` macro instead.

```diff
diff --git a/configure.ac b/configure.ac
index cbfe193..dcf37ad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,6 +10,7 @@ AC_CONFIG_AUX_DIR([config])
 AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([subdir-objects])
 AM_SILENT_RULES([yes])
+AM_CONDITIONAL([CO_INSTALLATION], [test "x$(FLUX_PREFIX)" = "x$(prefix)"])

 #LT_PREREQ([2.4.2])
 LT_INIT([dlopen])
diff --git a/t/Makefile.am b/t/Makefile.am
index 8e024e6..585b5ed 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -19,6 +19,12 @@ T_LOG_DRIVER = env AM_TAP_AWK='$(AWK)' $(SHELL) \
 # FLUX_SCHED_CO_INST is exported so that runlevel-install
 # test can conditionally set FLUX_RC_EXTRA.
 #
+if CO_INSTALLATION
+    CO_INST=co
+else
+    CO_INST=""
+endif
+
 TESTS_ENVIRONMENT = \
     LUA_PATH="$(FLUX_PREFIX)/share/lua/$(LUA_VERSION)/?.lua;$(LUA_PATH);;" \
     FLUX_LUA_PATH_PREPEND="$${FLUX_SCHED_TEST_INSTALLED:+$(prefix)/share/lua/$(LUA_VERSION)/?.lua}" \
@@ -27,7 +33,7 @@ TESTS_ENVIRONMENT = \
     FLUX_EXEC_PATH_PREPEND="$${FLUX_SCHED_TEST_INSTALLED:+$(prefix)/libexec/flux/cmd}" \
     FLUX_MODULE_PATH_PREPEND="$${FLUX_SCHED_TEST_INSTALLED:+$(prefix)/lib/flux/modules}" \
     FLUX_SCHED_RC_PATH="$${FLUX_SCHED_TEST_INSTALLED:+$(sysconf)/flux}" \
-    FLUX_SCHED_CO_INST=$(shell if [ $(FLUX_PREFIX) = $(prefix) ]; then echo co; fi) \
+    FLUX_SCHED_CO_INST=$(CO_INST) \
     PATH="$(FLUX_PREFIX)/bin:$(PATH)"
```

**Comment by dongahn** [comment]
Obviously, this doesn't work since `$(FLUX_PREFIX)` won't be available at configure time. Let me play with this some.

**Comment by dongahn** [comment]
Fixed in PR #405.
